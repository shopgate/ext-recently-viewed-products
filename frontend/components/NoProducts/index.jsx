import React, { useMemo } from 'react';
import I18n from '@shopgate/pwa-common/components/I18n';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';
import { encodeSVG } from '@shopgate/engage/core';
import styles from './style';

const { fullSVGs } = themeConfig;

/**
 * No products component
 * @returns {JSX.Element}
 */
const NoProducts = () => {
  const { noResultsImage } = fullSVGs;

  const imageSRC = useMemo(() => encodeSVG(noResultsImage),
    [noResultsImage]);

  return (
    <div className={styles.wrapper}>
      {noResultsImage && <img src={imageSRC} alt="" />}
      <I18n.Text string="recently_viewed_products.empty_product_list" />
    </div>
  );
};

export default NoProducts;
