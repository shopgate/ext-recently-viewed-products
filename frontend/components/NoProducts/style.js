import { css } from 'glamor';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const { variables } = themeConfig;

const wrapper = css({
  display: 'flex',
  height: '90vw',
  maxHeight: '40vh',
  minHeight: '90vw',
  textAlign: 'center',
  flexDirection: 'column',
  alignItems: 'center',
  justifyContent: 'center',
  margin: `${variables.gap.big}px ${variables.gap.big}px`,
  padding: `0 ${variables.gap.big}px ${variables.gap.big}px`,
});

export default {
  wrapper,
};
