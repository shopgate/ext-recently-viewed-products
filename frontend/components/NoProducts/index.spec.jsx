import React from 'react';
import { mount } from 'enzyme';
import NoProducts from './index';

describe('ProductList', () => {
  it('should render as expected', () => {
    const component = mount(<NoProducts />);

    expect(component).toMatchSnapshot();
  });
});
