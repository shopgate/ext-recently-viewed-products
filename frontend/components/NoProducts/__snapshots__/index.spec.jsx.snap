// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList should render as expected 1`] = `
<NoProducts>
  <div
    className={
      Object {
        "data-css-9gjavs": "",
      }
    }
  >
    <Translate
      className=""
      params={Object {}}
      string="recently_viewed_products.empty_product_list"
    >
      <span
        className=""
      >
        recently_viewed_products.empty_product_list
      </span>
    </Translate>
  </div>
</NoProducts>
`;
